lib/client/commands/login.go

```diff
@@ -79,23 +81,23 @@ func (l *Login) Run(cmd *cobra.Command, args []string) error {
 		}
 	}
 
-	client, err := l.base.AuthClient(l.rng)
+	conn, err := l.base.Connect()
 	if err != nil {
 		return err
 	}
-
-	options := auth.LoginOptions{
-		Context: l.base.Context(),
-		MinWait: l.MinWaitTime,
-		Store:   l.base.Local,
-	}
-
-	token, err := client.Login(username, domain, options)
+	repeater := retry.New(retry.WithWait(l.MinWaitTime), retry.WithRng(l.rng))
+	enCreds, err := kauth.PerformLogin(auth.NewAuthClient(conn), l.base.Log, repeater, username, domain)
 	if err != nil {
 		return err
 	}
+	if err := kauth.SaveCredentials(enCreds, l.base.Local, l.base.Log); err != nil && err != kauth.NoSSHCredentialsErr {
+		l.base.Log.Warnf("error saving credentials, err: %v", err)
+		return err
+	}
+
+	// TODO(adam): delete below when we are comfortable migrating from the token to pure ssh certificates
```

Comment:
I suspect this file is not going to go fully away. We still need to know the name of the user, so we don't have to ask it every time. I'd like the certificate of the remote server to be stored there as well, so we can eg, pin TLS or grpc auth to this certificate after login.

Reply:
Setting the user could be something entirely separate. IIRC we need the username for COOKIE_HOST and that should be it? Setting the name sort of like git config and performing actions based on the user-set config file seems okayish to me. 

Rough outline: NasshProxy.ServeCookie gets called. It blocks harder on authentication, Fragment target is now cred.Username + "@" + np.relayHost on authenticator success. 

If I set my username to Brad, tunnel expects Brad@nss-relay, but it fails  since I can't login as brad@host.

Reply:
We should chat, I've lost you on the last comment. Your identity should not be tied to what you're connecting to? you are adam@enfabrica.net when connecting to our proxy, @np.relayHost seems very suspicious to me.
If we add external guests support, they will be eg, bob@company.com, or foo@bar.biz?